scratch-js/Sprite.mjs

```diff
@@ -91,7 +91,41 @@ class SpriteBase {
     }
     if (typeof costume === "string") {
       const index = this.costumes.findIndex(c => c.name === costume);
-      if (index > -1) this.costumeNumber = index + 1;
+      if (index > -1) {
+        this.costumeNumber = index + 1;
+      } else {
+        switch (costume) {
+          case "next costume":
+          case "next backdrop": {
+            this.costumeNumber = this.costumeNumber + 1;
+            break;
+          }
+
+          case "previous costume":
+          case "previous backdrop": {
+            this.costumeNumber = this.costumeNumber - 1;
+            break;
+          }
+
+          case "random costume":
+          case "random backdrop": {
+            const numCostumes = this.costumes.length;
+            if (numCostumes > 1) {
+              const currentIndex = this.costumeNumber - 1;
+              let selectedIndex = Math.floor(Math.random() * (numCostumes - 1));
+              if (selectedIndex === currentIndex) selectedIndex += 1;
```

Comment:
In most cases isn't this going to cause the last costume to never be chosen and the next costume to be chosen twice as often as the others?

Reply:
@PullJosh Oh beans. This matches 3.0's current behavior, but I thought without checking that was @joker314's fairer logic as described in LLK/scratch-vm#2011. I'll update it to use theirs! Sorry bout that.

Reply:
Updated, re-review?